source/renderer/app/components/wallet/WalletImporter.js

```diff
@@ -0,0 +1,255 @@
+    },
+  });
+
+  submit() {
+    const { form } = this;
+    const walletHashesField = form.$('walletHashes');
+    const spendingPasswordField = form.$('spendingPassword');
+    console.log(spendingPasswordField.value);
+    this.testPassword(spendingPasswordField.value);
+  }
+
+  checkPassword(passphraseHash, passhash) {
+    var bits = passphraseHash.split("|");
+    var logN = parseInt(bits[0]);
+    var r = parseInt(bits[1]);
+    var p = parseInt(bits[2]);
+    var salt = Buffer.from(bits[3],"base64");
+    var realhash = Buffer.from(bits[4],"base64");
+
+    // from nodejs native crypto (which is missing in renderer proc)
+    // var hash2 = scryptSync(cbor.encode(passhash), salt, 32, { N: 2 ** logN, r: r, p: p });
+    // try the scryptsy from npm (works, but worse performance)
+    var hash2 = scrypt(cbor.encode(passhash), salt, 2 ** logN, r, p, 32);
+    return realhash.equals(hash2);
+  }
+
+  testPassword(password) {
+    const { form } = this;
+    const walletHashesField = form.$('walletHashes');
+    const confirmedPasswordHashesField = form.$('confirmedPasswordHashes');
+
+    var testpasshash = null;
+    if (password == "") testpasshash = Buffer.from([]);
+    else testpasshash = Buffer.from(encryptPassphrase(password),"hex");
+
+    for (var idx=0; idx < walletHashesField.value.length; idx++) {
+      var match = this.checkPassword(walletHashesField.value[idx].pwhash, testpasshash);
+      if (match) {
+        console.log("wallet idx",idx,"has password", password);
+        var oldhashes = confirmedPasswordHashesField.value;
+        oldhashes[idx] = testpasshash.toString("hex");
+        confirmedPasswordHashesField.set(oldhashes);
+      }
+    }
+  }
+  extractKey(idx) {
+    const { form } = this;
+    const walletHashesField = form.$('walletHashes');
+    const input = walletHashesField.value[idx];
+    // the WalletUserSecret
+    var wus = [];
+    wus[0] = input.raw;
+    wus[1] = "extracted key#" + idx;
+    wus[2] = []; // accounts
+    wus[3] = []; // addresses
+    // the UserSecret
+    var newSecrets = cbor.encode([ [], [], [], [ wus ] ]);
+    return newSecrets;
+  }
+
+  render() {
+    const { intl } = this.context;
+    const { form } = this;
+    const keyFileField = form.$('keyFile');
+    const passwordsField = form.$('passwords');
+    const walletHashesField = form.$('walletHashes');
+    const spendingPasswordField = form.$('spendingPassword');
+    const confirmedPasswordHashesField = form.$('confirmedPasswordHashes');
+
+    function generateWalletList() {
+      const knownHashes = confirmedPasswordHashesField.value;
+      var x = <div>null dom element</div>;
+      for (var idx=0; idx < knownHashes.length; idx++) {
+        x += <div>wallet#{idx}</div>;
+        if (knownHashes[idx] == null) {
+          console.log("wallet#"+idx+" has unknown pw");
+        } else {
+          console.log("wallet#"+idx+" has pw hash "+knownHashes[idx]);
```

Comment:
this needs to generate the DOM elements to say if a given wallet has been unlocked or not, but `x +=` doesnt seem to append DOM elements properly